numpy/core/src/multiarray/ctors.c

```diff
@@ -529,10 +529,11 @@ setArrayFromSequence(PyArrayObject *a, PyObject *s,
         return 0;
     }
 
-    slen = PySequence_Length(s);
-    if (slen < 0) {
+    PyObject *seq = PySequence_Fast(s, "Could not convert object to sequence");
```

Comment:
There are goto's above this definition (causing test failures, because `seq` may be uninitialized).